src/cpp/session/modules/build/SessionBuild.cpp

```diff
@@ -1185,7 +1185,18 @@ class Build : boost::noncopyable,
       std::string shinyTestName;
       if (type == kTestShinyFile) {
         shinyTestName = shinyPath.getFilename();
-        shinyPath = shinyPath.getParent().getParent();
+        shinyPath = shinyPath.getParent();
+        if (shinyPath.getFilename() == "shinytests")
+        {
+           // In newer versions of shinytest, tests are stored in a "shinytests" folder under the
+           // "tests" folder.
+           shinyPath = shinyPath.getParent();
+        }
+        if (shinyPath.getFilename() == "tests")
+        {
+           // Move up from the tests folder to the app folder.
+           shinyPath = shinyPath.getParent();
+        }
```

Comment:
Would it be worth inserting some warnings/logging under an `else` block? I'm hesitant to just push forward and hope for the best if the dir layout doesn't match what we expect.

Reply:
That's a good point. I've just added an else block that bails out when the test file doesn't look like it's actually in a tests dir.